language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -388,12 +388,13 @@ genDefDataType curPkgId conName mod tpls def =
                             ] ++
                             ["};"]
                         associatedTypes =
-                          let tT = conName
-                              tK = tT <> ".Key"
+                          let keyDef = maybe "unknown" (fst . genType (moduleName mod) . tplKeyType) (tplKey tpl)
+                              tT = conName
+                              tK = if keyDef == "unknown" then keyDef else tT <> ".Key"
                               tI = "typeof " <> tT <> ".templateId" in
-                          [ "export namespace " <> tT <> " {"
-                          , "  export type Key = " <> maybe "unknown" (fst . genType (moduleName mod) . tplKeyType) (tplKey tpl)
-                          , "  export type CreateEvent = damlLedger.CreateEvent" <> "<" <> tparams [tT, tK, tI] <> ">"
+                          [ "export namespace " <> tT <> " {" ] ++
+                          [ "  export type Key = " <> keyDef | keyDef /= "unknown" ] ++
```

Comment:
Matching on the string here seems a bit hacky, can we make `keyDef` a `Maybe` and then do something like `[ " … " <> keyDef | Just keyDef <- [ mbKeyDef ]]`?

Reply:
ok